src/AutoMapper/XpressionMapper/Extensions/MapperExtensions.cs

```diff
@@ -63,18 +92,14 @@ public static TDestDelegate MapExpressionAsInclude<TDestDelegate>(this IMapper m
             if (expression == null)
                 return null;
 
-            var typeSourceFunc = expression.GetType().GetGenericArguments()[0];
-            var typeDestFunc = typeof(TDestDelegate).GetGenericArguments()[0];
-
-            var typeMappings = new Dictionary<Type, Type>()
-                                            .AddTypeMappingsFromDelegates(typeSourceFunc, typeDestFunc);
-
-            XpressionMapperVisitor visitor = new MapIncludesVisitor(mapper == null ? Mapper.Configuration : mapper.ConfigurationProvider, typeMappings);
-            var remappedBody = visitor.Visit(expression.Body);
-            if (remappedBody == null)
-                throw new InvalidOperationException(Resource.cantRemapExpression);
-
-            return (TDestDelegate)Lambda(typeDestFunc, remappedBody, expression.GetDestinationParameterExpressions(visitor.InfoDictionary, typeMappings));
+            return mapper.MapExpression<TDestDelegate>
+            (
+                mapper == null ? Mapper.Configuration : mapper.ConfigurationProvider,
+                expression,
+                expression.GetType().GetGenericArguments()[0],
+                typeof(TDestDelegate).GetGenericArguments()[0],
```

Comment:
There is still some duplication here.

Reply:
Need to check for null before using expression.

Reply:
https://github.com/BlaiseD/AutoMapper/blob/fc4fe428f07bd695145624a3a82036d9a2d24d07/src/AutoMapper/XpressionMapper/Extensions/MapperExtensions.cs#L28